Frameworks/Foundation/NSIndexSet.mm

```diff
@@ -471,26 +498,44 @@ - (void)enumerateIndexesInRange:(NSRange)range options:(NSEnumerationOptions)opt
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
 + (BOOL)supportsSecureCoding {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return YES;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (id)initWithCoder:(NSCoder*)decoder {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (self = [super init]) {
+        NSUInteger length = 0;
+        const NSRange* ranges = [decoder isKindOfClass:[NSKeyedUnarchiver class]] ?
+                                    reinterpret_cast<const NSRange*>([decoder decodeBytesForKey:c_archiveKey returnedLength:&length]) :
+                                    reinterpret_cast<const NSRange*>([decoder decodeBytesWithReturnedLength:&length]);
+
+        if (ranges && length > 0) {
+            for (size_t i = 0; i < length / sizeof(NSRange); ++i) {
+                [static_cast<NSMutableIndexSet*>(self) _addItem:ranges[i]];
+            }
+        }
+    }
+    return self;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (void)encodeWithCoder:(NSCoder*)coder {
-    UNIMPLEMENTED();
+    unsigned int count = [static_cast<NSMutableIndexSet*>(self) _count];
+    if (count > 0) {
+        NSRangePointer ranges = [static_cast<NSMutableIndexSet*>(self) _allRanges];
+        if ([coder isKindOfClass:[NSKeyedArchiver class]]) {
+            [coder encodeBytes:reinterpret_cast<uint8_t*>(ranges) length:count * sizeof(NSRange) forKey:c_archiveKey];
+        } else {
```

Comment:
This will all but guarantee that range archives serialized on 32-bit platforms are unusable on 64-bit ones, and vice versa. As archives are intended for data portability, this may not be great.

Unfortunately, I don't have a proper solution here; we may want to discuss it.